### ISSUE TITLE
Godot 4 Branch always asks to save the project on open

### ISSUE THREAD

**Issue by MichaelJBrune** [quentincaffeino/godot-console#111]
With the GDScript demo, just open the project and close it. Godot will ask to save the project. Seems like something is changing in the plugin but I can't tell what yet. Will debug later but wanted to get an issue open so I don't forget.

**Comment by scriptsengineer** [comment]
I'm working on a fork (https://github.com/expressobits/godot-console) with several improvements (or not) that include
- Put godot 4 in a version that can be added as a submodule
- Add a required prefix to a command (Same minecraft)
- Separation of Console logic from UI (For UI to be customized by user)

My first commits of this journey fixed this unnecessary import for the addon.

**Comment by quentincaffeino** [comment]
Thank you for taking a look at it @scriptsengineer 

I don't like the idea of using addons as submodules but you do you :)

Alternatively I explored the option to use pnpm(npm) as a dependency manager for godot projects and it worked nicely.

What I was working lately on is packing godot as an npm dependency ehich would ease working with it a lot.

I still have some ideas and wip stuff regarding this project in branches but let's be honest, since it has been few years since I really used godot who knows when I'm going to get back to some bigger extent than couple commits a year.

**Comment by scriptsengineer** [comment]
Can we create a [discussion](https://github.com/quentincaffeino/godot-console/discussions) about npm/modules, wanted to know more about

**Comment by quentincaffeino** [comment]
> Can we create a [discussion](https://github.com/quentincaffeino/godot-console/discussions) about npm/modules, wanted to know more about

Sure, is something required form me or you'd post a discussion there?

**Comment by scriptsengineer** [comment]
Yes, I will create it soon, I'm just commenting because it might not be part of this issue

**Comment by MichaelJBrune** [comment]
> My first commits of this journey fixed this unnecessary import for the addon.

I wanted to get the most minimalistic fix, it looks like your first commits of that repo are moving it into a module rather than directly addressing the issue where the engine thinks something is changing. Do you have more data on what's changing and why?

**Comment by scriptsengineer** [comment]
It would be the first commits, a reimport solved it for me, as soon as I have time I can send the PR with just the correction.
